### PR TITLE
Update docs for >>=. and >>-.

### LINKED ISSUE #136: >>-. and >>=. do not behave as the documentation suggests
According to the documentation, the two examples below should have the same effect; in practice they do not due to eager evaluation of arguments in F#.

I'm not sure if it's possible to make Hopac behave as the docs describe, but if not it would be nice to update the documents to match reality :)

```fs
open Hopac
open Hopac.Infixes

// should wait, but doesn't
printfn "before sync %A" System.DateTime.Now
Job.result "hello"
>>-. printfn "before async %A" System.DateTime.Now
>>=. timeOutMillis 2000
>>-. printfn "after async %A" System.DateTime.Now // This prints immediately after "before async"
|> run
printfn "after sync %A" System.DateTime.Now

// should wait, and does
printfn "before sync %A" System.DateTime.Now
Job.result "hello"
>>- fun _ -> printfn "before async %A" System.DateTime.Now
>>= fun () -> timeOutMillis 2000
>>- fun () -> printfn "after async %A" System.DateTime.Now
|> run
printfn "after sync %A" System.DateTime.Now
```

### PR DESCRIPTION
**Pull request by neoeinstein**
Closes #136

/cc @mavnn

### CODE REVIEW

**Comment by mavnn** [comment]
Thanks Marcus, I hadn't managed to get this far

**Comment by neoeinstein** [comment]
What about changing this up a bit and acknowledging the eagerness earlier. Also bring in the earlier comment on `x -> 'y -> 'x

```fsharp
  /// Creates a job that runs the given two jobs and returns the result of the
  /// second job.  `xJ >>=. yJ` is equivalent to `xJ >>= fun _ -> yJ` when `yJ`
  /// is a value. If `yJ` is an expression, side-effects from evaluating `yJ` to
  /// a `Job<'y> are not delayed.
  ///
  /// To delay evaluation of `yJ` when it is an expression with side-effects,
  /// use the full form `xJ >>= fun _ -> yJ`.
#if DOC
  ///
  /// As an example, evaluating the first line will print "worldhello", while the
  /// second will print "helloworld":
  ///
  ///> Job.thunk (fun () -> printf "hello") >>=. Job.result (printf "world")
  ///> Job.thunk (fun () -> printf "hello") >>= fun _ -> Job.result (printf "world")
#endif
  val ( >>=.  ): Job<_>  ->         Job<'y>  ->     Job<'y>
```

**Comment by polytypic** [comment]
Note that `>>=.` and `>>-.` aren't the only places where the documentation is (semi-intentionally) sloppy.  For example, `^=>.` and `^->.` use similar equations.  Consider the following:

> `xA ^=>. yJ` is equivalent to `xA ^=> fun _ -> yJ`.

Now, it is clear that on the left hand side, `xA ^=>. yJ`, any effects from the evaluation of `xA` and `yJ` (if we interpret them as *metavariables*) will occur before the call of `^=>.`.  On the right hand side, `xA ^=> fun _ -> yJ`, any effects from `yJ` will not occure before the call of `^=>.`.

The above equation, and other equations like that, could also be changed to just evaluate `yJ` on the right hand side.  For example, the documentation could be changed to say

> `xA ^=>. yJ` is equivalent to `xA ^=> let yJ = yJ in fun _ -> yJ`.

or, for example,

> `xA ^=>. yJ` is equivalent to `xA ^=> always yJ`.

where `always` is the K combinator defined as

```fsharp
let always x _ = x
```

in the beginning of the documentation.  This way both sides of the equation would be equivalent with respect to side effects.

Here is another take.  Let's consider the equation once more:

> `xA ^=>. yJ` is equivalent to `xA ^=> fun _ -> yJ`.

What if we interpret that as a reference implementation?  The equation translates directly to concrete F#

```fsharp
let (^=>.) xA yJ = xA ^=> fun _ -> yJ
```

which is also just correct.  Note that here we are (indeed) treating `xA` and `yJ` as *variables* rather than as *metavariables*.

**Comment by neoeinstein** [comment]
I like using the `always` construct. I've added a preface up near the top of the documentation that introduces the `always` function, and have changed the rest of the documentation below.

I think that using `always` makes it more clear that `yJ`/`y` is intended to be a value, and thus allows us to reduce the amount of explanation that these four operators require.

File: Libs/Hopac/Hopac.fsi [hunk 1]
```diff
@@ -2626,16 +2635,30 @@ module Infixes =
...
 
-  /// `xA ^->. y` is equivalent to `xA ^-> fun _ -> y`.
+  /// Creates an alternative which is committed to when the given alternative
+  /// is committed to. Once committed, the given value is used as the result.
+  ///
+  /// `xA ^->. y` is equivalent to `xA ^-> always y`.
   val ( ^->. ): Alt<_>  ->             'y   -> Alt<'y>
 
+  /// Creates an alternative which is committed to when the alternative
+  /// argument is committed to. Once committed, the given exception is raised.
```
Comment (hyprhare) [inline Libs/Hopac/Hopac.fsi]: Unify phrasing with `ln 2654` for consistency.

**Pull request merged**